EtageControl/src/etage-control.ts

```diff
@@ -1,142 +1,165 @@
 import L from 'leaflet'
 import './etage-control.css'
 
+interface Building {
+  id: string
+  name: string
+  center: [number, number] // Latitude and longitude of the building
+  zoom: number
+}
+
+interface Level {
+  id: string
+  name: string
+  buildingId: string
+  center: [number, number] // We need this right now to use the same methods of the active option in the menu
+  zoom: number
+}
+
+// Using proper TS types for the options, including the custom ones.
+// To be verified.
+interface PlantquestEtageControlOptions extends L.ControlOptions {
+  buildings?: Building[]
+  levels?: Level[]
+}
+
 const PlantquestEtageControl = L.Control.extend({
   options: {
-    position: 'topright'
-  },
+    position: 'topright' // Default position, see this after in case of overlapping
+  } as PlantquestEtageControlOptions,
 
-  initialize: function (this: any, options: any) {
-    console.log('PGS init')
-    this._state = {}
+  // Initialize the properties of the class, including the custom ones.
+  _map: null as L.Map | null,
+  _buildings: null as Building[] | null,
+  _levels: null as Level[] | null,
+
+  initialize: function (options: PlantquestEtageControlOptions) {
     L.Util.setOptions(this, options)
```

Comment:
Use https://github.com/rjrodger/gubu for validation as per https://github.com/plantquest/leaflet-plugins/pull/8

Reply:
Done. Using it before, needs one more review, I'm new to Gubu.